mod_clamav.c

```diff
@@ -117,6 +117,12 @@ static int clamavd_result(int sockd, const char *abs_filename, const char *rel_f
      */
     pr_event_generate("mod_clamav.virus-found", pt);
 
+    if (pr_table_add(session.notes, "mod_clamav.virus-found",
+        pstrdup(session.pool, pt), 0) < 0) {
```

Comment:
Why use `session.notes` here, rather than the `cmd->notes` table for the current command?  That is, conceptually the detected virus is specific the file that was just uploaded (thus the `STOR` cmd), not to the entire session.  Over the course of a session, there could be multiple files uploaded, each file having a different detected virus.

For this reason, would it make sense to pass that `cmd_rec *cmd` into this function, for storing this in its notes, rather than the session-wide `notes` table?

Reply:
I don't know how that would be done here.  clamavd_result is called from either clamavd_scan or clamavd_scan_stream, which are called from clamav_fsio_close, which is registered as the fs->close handler for mod_clamav.  I'm assuming that the API for a fs->close handler is fixed and it doesn't appear to include cmd, so unless cmd is exposed via some other method, I don't believe that it can be used to store the virus name there.

That said, I believe that the way I did things should handle multiple uploads, as the virus name is stored during the file upload CMD, and upon completion of the upload/scan it is removed when the LOG_CMD_ERR handler fires - which I believe would happen before any subsequent uploads happened.  If that's not the case, then stashing it in the cmd->notes would certainly make more sense, assuming it can be accessed (but I don't know enough about the proftpd internals to know how to do that in this case).